subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/SkipCachedTaskExecuter.java

```diff
@@ -81,8 +89,7 @@ public void execute(final TaskInternal task, TaskStateInternal state, TaskExecut
                         cacheable = true;
                         TaskArtifactState taskState = context.getTaskArtifactState();
                         try {
-                            cacheKey = taskState.calculateCacheKey();
-                            LOGGER.info("Cache key for {} is {}", task, cacheKey);
+                            cacheKey = cacheKeyCalculator.calculate(taskState.getCurrentExecution(), task);
```

Comment:
We also need to make this so we're always creating the cache key for all tasks and all the time, not just the ones that have caching enabled.